presto-hive/src/main/java/io/prestosql/plugin/hive/authentication/HiveContext.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.authentication;
+
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.security.ConnectorIdentity;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public final class HiveContext
```

Comment:
I thought about this as well.

`HdfsContext` is named `context` today because it contains contextual information about what is being accessed, and provides that to the file system for logging, auditing, accounting, etc.

I can see wanting to add similar information into `HiveContext` so that the metastore could be aware of the Presto query ID, etc. This comes into play with caching -- if we are providing such information, we probably should be caching at the same granularity. However, while this seems somewhat orthogonal to impersonation, it's also intertwined.

Reply:
@electrum we cannot extend this class this way, exactly because we're using this class as a key in caches.
This is actually why it _should be named_ `HiveIdentity`.

If we are to add additional context information (eg Presto query ID), we will need a separate context object to pass this.